### ISSUE TITLE
Foreign key constraint violation when targets.(Service).deleteFromRepo

### ISSUE THREAD

**Issue by jakubbujny** [hashicorp/boundary#2064]
**Describe the bug**
Boundary v0.8.0
`boundary targets delete -id XXX`
gives

```
Error from controller when performing delete on target

Error information:
  Kind:                Internal
  Message:             targets.(Service).deleteFromRepo: unable to delete target: target.(Repository).DeleteTarget: db.DoTx: target.(Repository).DeleteTarget: db.Delete: insert or update on table "session_state" violates foreign key constraint
  "session_valid_state_enm_fkey": integrity violation: error #1003
  Status:              500
  context:             Error from controller when performing delete on target
```

**To Reproduce**
I'm not sure how to reproduce it exactly - we did the upgrade from 0.7.0 to 0.8.0
It seems only some part of targets is affected (these created in 0.7.0) - when I create new "test" target I'm able to delete it.

**Comment by jakubbujny** [comment]
I found now that we have the same problem for the host-sets

```
boundary host-sets delete -id hsst_q5cqOuwUja
Error from controller when performing delete on host set

Error information:
  Kind:                Internal
  Message:             host_sets.(Service).deleteFromRepo: unable to delete host: static.(Repository).DeleteSet: delete failed for hsst_q5cqOuwUja: db.DoTx: static.(Repository).DeleteSet: db.Delete: insert or update on table "session_state"
  violates foreign key constraint "session_valid_state_enm_fkey": integrity violation: error #1003
  Status:              500
  context:             Error from controller when performing delete on host set
  ```

**Comment by irenarindos** [comment]
Thanks for reporting this! I've been able to recreate it & am working on a fix.

**Comment by irenarindos** [comment]
Thanks again- this has been fixed in #2065 and will be in our next release.
